### PR TITLE
[Misc] Add python format script

### PR DESCRIPTION
**Pull request by brosoul**
## Pull Request Description
Add python format script, and add ruff lint and format rules.

**Before merge, it need to rebase `main` branch and run the`./scripts/format.sh` to do lint and format.**

## Related Issues
Resolves: https://github.com/aibrix/aibrix/issues/351

**Important: Before submitting, please complete the description above and review the checklist below.**

---

<details>
<summary><strong>Contribution Guidelines (Expand for Details)</strong></summary>

<p>We appreciate your contribution to aibrix! To ensure a smooth review process and maintain high code quality, please adhere to the following guidelines:</p>

<h3>Pull Request Title Format</h3>
<p>Your PR title should start with one of these prefixes to indicate the nature of the change:</p>
<ul>
    <li><code>[Bug]</code>: Corrections to existing functionality</li>
    <li><code>[CI]</code>: Changes to build process or CI pipeline</li>
    <li><code>[Docs]</code>: Updates or additions to documentation</li>
    <li><code>[API]</code>: Modifications to aibrix's API or interface</li>
    <li><code>[CLI]</code>: Changes or additions to the Command Line Interface</li>
    <li><code>[Misc]</code>: For changes not covered above (use sparingly)</li>
</ul>
<p><em>Note: For changes spanning multiple categories, use multiple prefixes in order of importance.</em></p>

<h3>Submission Checklist</h3>
<ul>
    <li>[ ] PR title includes appropriate prefix(es)</li>
    <li>[ ] Changes are clearly explained in the PR description</li>
    <li>[ ] New and existing tests pass successfully</li>
    <li>[ ] Code adheres to project style and best practices</li>
    <li>[ ] Documentation updated to reflect changes (if applicable)</li>
    <li>[ ] Thorough testing completed, no regressions introduced</li>
</ul>

<p>By submitting this PR, you confirm that you've read these guidelines and your changes align with the project's contribution standards.</p>

</details>

### CODE REVIEW

File: python/aibrix/scripts/format.sh [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        echo "$1 is not installed, please run \`poetry install --no-root --with dev\`"
+        exit 1
+    fi
+}
+
+check_command ruff
+check_command mypy
+
+# Run Ruff
+echo 'vLLM ruff (lint):'
```
Comment (Jeffwan) [inline python/aibrix/scripts/format.sh]: let's move `vLLM` here

File: python/aibrix/README.md [hunk 2]
```diff
@@ -25,7 +25,27 @@ At present, `vLLM` engine is supported, and in the future, `SGLang` and other in
...
+
+Before contribute your code, please run the following commands to ensure that your code passes the tests and linting checks.
+
+```bash
+# install dependencies
+poetry install --no-root --with dev
+
+# linting, formatting and type checking
+bash ./scripts/format.sh
+
```
Comment (Jeffwan) [inline python/aibrix/README.md]: nit: remove empty lines here

File: python/aibrix/scripts/format.sh [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+builtin cd "$ROOT" || exit 1
+
+check_command() {
+    if ! command -v "$1" &> /dev/null; then
+        echo "$1 is not installed, please run \`poetry install --no-root --with dev\`"
+        exit 1
+    fi
+}
+
+check_command ruff
```
Comment (Jeffwan) [inline python/aibrix/scripts/format.sh:24]: do we add the dependencies to the yml?  Let's pin a version to avoid the format conflicts between version

Comment (brosoul) [reply python/aibrix/scripts/format.sh:24]: resolved all the comments, please take a look cc @Jeffwan

**Pull request merged**